File: backend/module/Radio/src/Radio/Entity/Tag.php
<?php

namespace Radio\Entity;

use Doctrine\ORM\Mapping as ORM;

/**
 * @ORM\Entity
 * @ORM\Table(name="tag")
 * */
class Tag
{
    public static $PERSON = 1;
    public static $GENERIC = 0;

    /**
     * @ORM\Id
     * @ORM\Column(type="integer")
     * @ORM\GeneratedValue
     * */
    protected $id;

    /**
     * @ORM\Column(type="integer")
     * */
    protected $type = 0;

    /**
     * @ORM\Column(type="string")
     * */
    protected $name;

    /**
     * @ORM\ManyToMany(targetEntity="TextContent", mappedBy="tags")
     **/
    protected $textcontents;

    /**
     * @return mixed
     */
    public function getTextcontents()
    {
        return $this->textcontents;
    }

    /**
     * @param mixed $textcontents
     */
    public function setTextcontents($textcontents)
    {
        $this->textcontents = $textcontents;
    }


    /**
     * @return mixed
     */
    public function getId()
    {
        return $this->id;
    }

    /**
     * @param mixed $id
     */
    public function setId($id)
    {
        $this->id = $id;
    }

    /**
     * @return mixed
     */
    public function getName()
    {
        return $this->name;
    }

    /**
     * @param mixed $name
     */
    public function setName($name)
    {
        $this->name = $name;
    }

    /**
     * @return mixed
     */
    public function getType()
    {
        return $this->type;
    }

    /**
     * @param mixed $type
     */
    public function setType($type)
    {
        $this->type = $type;
    }


}

?>

File: util/model/src/main/java/hu/radio/tilos/model/Tag.java
package hu.radio.tilos.model;

import hu.radio.tilos.model.type.MixCategory;
import hu.radio.tilos.model.type.MixType;
import hu.radio.tilos.model.type.TagType;

import javax.persistence.*;
import java.util.ArrayList;
import java.util.Date;
import java.util.List;

@Entity()
@Table(name = "tag")
@NamedQuery(name = "tag.byName", query = "SELECT t FROM Tag t WHERE t.name = :name")
public class Tag {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;

    @Basic
    @Column(length = 255)
    private String name;

    @Column
    TagType type;

    @ManyToMany
    @JoinTable(
            name = "tag_textcontent",
            joinColumns = {@JoinColumn(name = "tag_id", referencedColumnName = "id")},
            inverseJoinColumns = {@JoinColumn(name = "textcontent_id", referencedColumnName = "id")})
    private List<TextContent> taggedTexts = new ArrayList<TextContent>();

    public List<TextContent> getTaggedTexts() {
        return taggedTexts;
    }

    public void setTaggedTexts(List<TextContent> taggedTexts) {
        this.taggedTexts = taggedTexts;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public TagType getType() {
        return type;
    }

    public void setType(TagType type) {
        this.type = type;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (!(o instanceof Tag)) return false;

        Tag tag = (Tag) o;

        if (id != tag.id) return false;
        if (!name.equals(tag.name)) return false;
        if (type != tag.type) return false;

        return true;
    }

    @Override
    public int hashCode() {
        int result = id;
        result = 31 * result + name.hashCode();
        result = 31 * result + type.hashCode();
        return result;
    }
}
